File: Programmers/Level1/키패드누르기/키패드누르기.js
function solution(numbers, hand) {
    let answer = "";
    const keypad = {
        1: [0, 0],
        2: [0, 1],
        3: [0, 2],
        4: [1, 0],
        5: [1, 1],
        6: [1, 2],
        7: [2, 0],
        8: [2, 1],
        9: [2, 2],
        "*": [3, 0],
        0: [3, 1],
        "#": [3, 2],
    };

    let left_Hand = "*";
    let right_Hand = "#";

    for (let num of numbers) {
        let cur = keypad[num];
        let left_Position = keypad[left_Hand];
        let right_Position = keypad[right_Hand];

        if ([1, 4, 7].includes(num)) {
            answer += "L";
            left_Hand = num;
        } else if ([3, 6, 9].includes(num)) {
            answer += "R";
            right_Hand = num;
        } else {
            left_Dist =
                Math.abs(cur[0] - left_Position[0]) +
                Math.abs(cur[1] - left_Position[1]);
            right_Dist =
                Math.abs(cur[0] - right_Position[0]) +
                Math.abs(cur[1] - right_Position[1]);

            if (left_Dist < right_Dist) {
                left_Hand = num;
                answer += "L";
            } else if (left_Dist > right_Dist) {
                right_Hand = num;
                answer += "R";
            } else {
                if (hand == "right") {
                    right_Hand = num;
                    answer += "R";
                } else {
                    left_Hand = num;
                    answer += "L";
                }
            }
        }
    }
    return answer;
}

File: Programmers/Level1/키패드누르기/키패드누르기.py
def solution(numbers, hand):
    answer = ''
    keyPad = {
        1: (0, 0), 2: (0, 1), 3: (0, 2), 
        4: (1, 0), 5: (1, 1), 6: (1, 2),
        7: (2, 0), 8: (2, 1), 9: (2, 2),
        '*': (3, 0), 0: (3, 1), '#': (3, 2) 
    }
    left = [1, 4, 7]
    right = [3, 6, 9]
    l_hand = '*'
    r_hand = '#'
    
    for i in numbers:
        if i in left:
            answer += 'L'
            l_hand = i
        elif i in right:
            answer += 'R'
            r_hand = i
        else:
            cur = keyPad[i]
            l_position = keyPad[l_hand]
            r_position = keyPad[r_hand]
            l_dist = abs(cur[0] - l_position[0]) + abs(cur[1] - l_position[1])
            r_dist = abs(cur[0] - r_position[0]) + abs(cur[1] - r_position[1])
            
            if l_dist > r_dist:
                r_hand = i
                answer += 'R'
            elif l_dist < r_dist:
                l_hand = i
                answer += 'L'
            else:
                if hand =='right':
                    r_hand = i
                    answer += 'R'
                else:
                    l_hand = i
                    answer += 'L'
                    
    return answer